module/synchronization/core.go

```diff
@@ -51,19 +51,23 @@ func DefaultConfig() Config {
 //
 // Core is safe for concurrent use by multiple goroutines.
 type Core struct {
-	log      zerolog.Logger
-	Config   Config
-	mu       sync.Mutex
-	heights  map[uint64]*Status
-	blockIDs map[flow.Identifier]*Status
+	log                  zerolog.Logger
+	Config               Config
+	mu                   sync.Mutex
+	heights              map[uint64]*Status
+	blockIDs             map[flow.Identifier]*Status
+	metrics              SynchronizationMetrics
+	localFinalizedHeight int
 }
 
-func New(log zerolog.Logger, config Config) (*Core, error) {
+func New(log zerolog.Logger, config Config, metrics SynchronizationMetrics) (*Core, error) {
 	core := &Core{
-		log:      log.With().Str("module", "synchronization").Logger(),
-		Config:   config,
-		heights:  make(map[uint64]*Status),
-		blockIDs: make(map[flow.Identifier]*Status),
+		log:                  log.With().Str("module", "synchronization").Logger(),
+		Config:               config,
+		heights:              make(map[uint64]*Status),
+		blockIDs:             make(map[flow.Identifier]*Status),
+		metrics:              metrics,
+		localFinalizedHeight: -1, // used in ScanPending
```

Comment:
Could we use `uint64` (same type as `Header.Height`) and initialize it to `0` instead?